test/functionalTest/cases/2995_q_in_subscriptions/q_in_subscriptions.test

```diff
@@ -0,0 +1,237 @@
+# Copyright 2017 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+Q filter in subscriptions
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+accumulatorStart --pretty-print
+
+--SHELL--
+
+#
+# 01. Create subscription for entity E, attribute A with a filter q="t.'a.b'.'c.d'>=20"
+# 02. Create entity E1 with t.a.b.c.d == 20
+# 03. Make sure no notification has been sent
+# 04. Create entity E2 with t.'a.b'.'c.d' == 19
+# 05. Make sure no notification has been sent
+# 06. Create entity E3 with t.'a.b'.'c.d' == 20
+# 07. Make sure ONE notification has been sent
+#
+
+echo "01. Create subscription for entity E, attribute t with a filter q=t.'a.b'.'c.d'>=20"
+echo "==================================================================================="
+filter="t.'a.b'.'c.d'>=20"
+payload='{
+  "subject": {
+    "entities": [
+      {
+        "idPattern": "E.*"
+      }
+    ],
+    "condition": {
+      "attrs": [ "t" ],
+      "expression": { "q": "'$filter'" }
```

Comment:
I'd suggest to add a new .test with the same steps script, but using mq that time.

Reply:
The entire battery of func tests added in f3ad37f and 16af92f